lib/dry/validation/schema.rb

```diff
@@ -92,6 +94,14 @@ def [](name)
       def predicates
         self.class.predicates
       end
+
+      private
+
+      def validate_input(input)
+        unless input.is_a?(Hash)
+          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
+        end
```

Comment:
@solnic That may be handy, however it would probably lead to ISP violations if you had one schema defining how to validate multiple different objects with a schema like:

```ruby
class Schema < Dry::Validation::Schema
  hash? do
    attr(:name, &:filled?)
  end

  entity? do
    attr(:name, &:filled?)
  end
end
```

I could also be misunderstanding what you are describing.